examples/multifit/multifit.py

```diff
@@ -0,0 +1,129 @@
+
+To get a hold on this nonlinear behavior, the model must take
+the temperature of the resistor into account. Thus, the
+temperature is also recorded for every data point.
+The data thus consists of triples, instead of the usual "xy" pairs,
+and the relationship between temperature and voltage must be
+modeled in addition to the one between current and voltage.
+
+Here, the dependence :math:`T(U)` is taken to be quadratic, with
+some coefficients :math:`p_0`, :math:`p_1`, and :math:`p_2`:
+
+.. math::
+
+    T(U) = p_2 U^2 + p_1 U + p_0
+
+This model is based purely on empirical observations. The :math:`I(U)`
+dependence is more complicated, but takes the "running" of the
+resistane with the temperature into account:
+
+.. math::
+
+    I(U) = \frac{U}{R_0 (1 + t \cdot \alpha_T)}
+
+In the above, :math:`t` is the temperature in degrees Celsius,
+:math:`\alpha_T` is an empirical "heat coefficient", and :math:`R_0`
+is the resistance at 0 degrees Celsius, which we want to determine.
+
+In essence, there are two models here which must be fitted to the
+:math:`I(U)` and :math:`T(U)` data sets, and one model "incorporates"
+the other in some way.
+
+
+Approach 2: multi-model fit
+---------------------------
+
+There are several ways to achieve this with *kafe2*. The method chosen
+here is to use the :py:object:`~kafe.fit.xy_multi.XYMultifit` functionality
+to fit both models simultaneously to the :math:`T(U)` and :math:`I(U)`
+datasets.
+
+In general, this approach yields different results than the one using
+parameter constraints, which is demonstrated in the example called constraints.
+"""
+
+
+from kafe.fit import XYMultiFit, XYMultiPlot
+import numpy as np
+import matplotlib.pyplot as plt
+
+# empirical model for T(U): a parabola
+# independent variable MUST be named x!
+def empirical_T_U_model(x, p2=1.0, p1=1.0, p0=0.0):
+    # use quadratic model as empirical temerature dependence T(U)
+    return p2 * x**2 + p1 * x + p0
+
+
+
+# model of current-voltage dependence I(U) for a heating resistor
+# independent variable MUST be named x!
+def I_U_model(x, R0=1., alph=0.004, p2=1.0, p1=1.0, p0=0.0):
+    # use quadratic model as empirical temerature dependence T(U)
+    _temperature = empirical_T_U_model(x, p2, p1, p0)
+    # plug the temperature into the model
+    return x / (R0 * (1.0 + _temperature * alph))
+
+
+# -- Next, read the data from an external file
+
+# load all data into numpy arrays
+U, I, T = np.loadtxt('OhmsLawExperiment.dat', unpack=True)  # data
+sigU, sigI, sigT = 0.1, 0.1, 0.1  # uncertainties
+
+T0 = 273.15  # 0 degrees C as absolute Temperature (in Kelvin)
+T -= T0 #Measurements are in Kelvin, convert to °C
+
+# -- Finally, go through the fitting procedure
+
+# Step 1:  construct an XYMultiFit object
+fit = XYMultiFit(xy_data=[[U, T], [U, I]],
+                 model_function=[empirical_T_U_model, I_U_model])
+
+# declare errors on U
+fit.add_simple_shared_x_error(err_val=sigU, correlation=0, relative=False)

```

Comment:
I didn't add parameters to `add_simple_error` because the thought didn't occur to me.
I thought about `add_simple_shared_x_error` as a special case of `add_matrix_error`, though technically the same is true for `add_simple_error`.

Regardless, I think that there should be a separate function.
The way I understand it the functionality for shared errors only makes sense if they're applied to all models and if they're applied to the x-axis.
I think that it's better to have a separate method where those parameters are set to the appropriate values than to add checks to a larger method to check if all parameters are consistent.